tools/commit_slider/case_1.md

```diff
@@ -0,0 +1,316 @@
+
+## Third Version: Specific Template
+###  API design
+We want to provide for user 2 ways to set up command lines for comparation. ```'appCmd' : ['./benchmark_app --parameters_1', './benchmark_app --parameters_2']``` for detailed parameters and simplified ```'par_1' : 'parameter_1', 'par_2' : 'parameter_2'``` supposing that substitute the placeholder ```'appCmd' : 'prefix {actualPar} postfix'```. The example of this approach may be ```'par_1' : 'model_1.xml', 'par_2' : 'model_2.xml' : 'parameter_2', 'appCmd' : './benchmark_app -m {actualPar} -i input.png'```
+  
+### Custom template
+```
+from  utils.templates.common_template  import  Template
+
+class  BenchmarkCrossCheckTemplate(Template):
+
+def  printResult(commitPath, outLogger, getCommitInfo):
+    # formatting of output anf post analisys
+    print("\n" + "*" * 17 "<Commit slider output>" + "*" * 17 "\nTable of throughputs:" + '\n\t\thr_1\thr_2')
+    for  pathcommit  in  commitPath.getList():
+        commitInfo = getCommitInfo(pathcommit)
+        cHash = pathcommit.cHash
+        cHash = cHash[:7]
+        commMsg = "{}\t\t{}\t{}".format( cHash,
+            pathcommit.firstThroughput,
+            pathcommit.secondThroughput)
+        print(commMsg)
+        outLogger.info(commitInfo)
+    cPath = commitPath.getList()
+    thr00 = float(cPath[0].firstThroughput)
+    thr01 = float(cPath[0].secondThroughput)
+    thr10 = float(cPath[1].firstThroughput)
+    
+    print("{}\nSupposed rootcause is: {}".format(
+        "*" * 58,
+        'Model'  if  abs(thr00 - thr01) > abs(thr00 - thr10) else  'OV'
+        ))
+    print("*" * 58" + "\n")
+    print("* m1 = {}".format(pathcommit.firstModel))
+    print("** m2 = {}".format(pathcommit.secondModel))
+
+def  generateFullConfig(cfg, customCfg):
+    tmpJSON = cfg
+    # handling input commits
+    if  'c'  in  customCfg:
+        curCfg = tmpJSON['runConfig']
+        interval = customCfg['c'].split("..")
+        start, end = interval[0], interval[1]
+        curCfg['commitList'] = {'explicitList': [ start, end ] }
+    tmpJSON['runConfig'] = curCfg
+    # passing base parameters
+    Template.passParameters([
+       'gitPath', 'buildPath', 'appPath'
+        ], customCfg, tmpJSON)
+    # resolving of API types
+    if  isinstance(customCfg['appCmd'], list):
+        curCfg = tmpJSON['runConfig']
+        curCfg['par_1'] = customCfg['appCmd'][0]
+        curCfg['par_2'] = customCfg['appCmd'][1]
+        tmpJSON['runConfig'] = curCfg
+    else:
+        curCfg = tmpJSON['runConfig']
+        curCfg['par_1'] = customCfg['appCmd'].format(model=customCfg['modelList'][0])
+        curCfg['par_2'] = customCfg['appCmd'].format(model=customCfg['modelList'][1])   
+        tmpJSON['runConfig'] = curCfg
+    tmpJSON['appCmd'] = "{actualPar}"
+```
+---
+### Configuration and CLI
+First API
```

Comment:
Please explain what "First" and "Second" API mean

Reply:
Done